tests/dist/python/run_dist_objects.py

```diff
@@ -48,6 +58,154 @@ def run_server(
+        assert g.number_of_nodes(key) == n_nodes
+        node_split_test(g, force_even=False, ntype=key)
+        node_split_test(g, force_even=True,  ntype=key)
+
+    for key, n_edges in num_edges.items():
+        assert g.number_of_edges(key) == n_edges
+        edge_split_test(g, force_even=False, etype=key)
+        edge_split_test(g, force_even=True,  etype=key)
+
+##########################################
+########### DistGraphServices ###########
+##########################################
+    nids = F.arange(0, 16)
+
+    # Test in_degrees
+    orig_in_degrees = g.ndata['in_degrees']
+    local_in_degrees = g.in_degrees(nids)
+    F.allclose(local_in_degrees, orig_in_degrees[nids])
+
+    # Test out_degrees
+    orig_out_degrees = g.ndata['out_degrees']
+    local_out_degrees = g.out_degrees(nids)
+    F.allclose(local_out_degrees, orig_out_degrees[nids])
+
+    find_edges_test(g)
+    edge_subgraph_test(g)
+    sample_neighbors_test(g)
+
+def find_edges_test(g, orig_nid_map):
+    etypes = g.canonical_etypes
+
+    etype_eids_uv_map = dict()
+    for u_type, etype, v_type in etypes:
+        orig_u =  g.edges[etype].data['edge_u']
+        orig_v =  g.edges[etype].data['edge_v']
+        eids = F.tensor(np.random.randint(g.number_of_edges(etype), size=100))
+        u, v = g.find_edges(eids, etype=etype)
+        assert F.allclose(orig_nid_map[u_type][u], orig_u[eids])
+        assert F.allclose(orig_nid_map[v_type][v], orig_v[eids])
+        etype_eids_uv_map[etype] = (eids, F.cat([u, v], dim=0))
+    return etype_eids_uv_map
+
+def edge_subgraph_test(g, etype_eids_uv_map):
+    etypes = g.canonical_etypes
+    all_eids = dict()
+    for t in etypes:
+        all_eids[t] = etype_eids_uv_map[t[1]][0]
+
+    sg = g.edge_subgraph(all_eids)
+    for t in etypes:
+        assert sg.number_of_edges(t[1]) == len(all_eids[t])
+        assert F.allclose(sg.edges[t].data[dgl.EID], all_eids[t])
+
+    for u_type, etype, v_type in etypes:
+        uv = etype_eids_uv_map[etype][1]
+        sg_u_nids = sg.nodes[u_type].data[dgl.NID]
+        sg_v_nids = sg.nodes[v_type].data[dgl.NID]
+        sg_uv = F.cat([sg_u_nids, sg_v_nids], dim=0)
+        for node_id in uv:
+            assert node_id in sg_uv
+
+def sample_neighbors_with_args(g, size, fanout):
+    num_nodes = {ntype : g.num_nodes(ntype) for ntype in g.ntypes}
+    etypes = g.canonical_etypes
+
+    sampled_graph = g.sample_neighbors({ntype : np.random.randint(0, n, size=size) for ntype, n in num_nodes.items()}, fanout)
+
+    for ntype, n in num_nodes.items():
+        assert sampled_graph.number_of_nodes(ntype) == n
+    for t in etypes:
+        src, dst = sampled_graph.edges(etype=t)
+        eids = sampled_graph.edges[t].data[dgl.EID]
+        dist_u, dist_v = g.find_edges(eids, etype=t[1])
+        assert F.allclose(dist_u, src)
+        assert F.allclose(dist_v, dst)
+
+def sample_neighbors_test(g):
+    sample_neighbors_with_args(g, size=5, fanout=3)
```

Comment:
the `size` here corresponds the `batch_size`, let's test with larger ones such as `1024`, `10*1024`.